hydromt/raster.py

```diff
@@ -2233,6 +2234,243 @@ def tile_window(shape, px):
         with open(join(root, f"{mName}.yml"), "w") as f:
             yaml.dump({mName: yml}, f, default_flow_style=False, sort_keys=False)
 
+    def to_osm_tiles(
+        self,
+        root: str,
+        zoom_levels: list,
+        bbox: tuple = (),
+        method: str = "nearest",
+        min_lvl: int = 5,
```

Comment:
Would it be possible to calculate min_lvl from the original data, similar to to_xyz_tiles?

Reply:
Yes, that's what dres and nzl are supposed to do.  (line 2304-2306).
But at this moment in time, it's badly written. Apperently it also causes issues using more local references (which was to be expected)
I'll get busy on writing a proper implementation.

@DirkEilander, send me the dataset you used so I can test it.

Reply:
This is now hopefully fixed.

Reply:
It's more like a check now for a warning now, but we could later on chose to not use the min_lvl kwargs anymore.